src/front/src/DocumentPreview.js

```diff
@@ -0,0 +1,44 @@
+import {useEffect, useState} from 'react';
+import {Document, Page, pdfjs} from 'react-pdf';
+import "./DocumentPreview.css";
+
+pdfjs.GlobalWorkerOptions.workerSrc = `//unpkg.com/pdfjs-dist@${pdfjs.version}/build/pdf.worker.min.js`;
+
+/** @type {(document: Blob) => boolean} */
+export const isPdf = document => ['application/pdf', 'application/x-pdf'].includes(document.type);
+
+const plainTextSet = new Set([
+    'bsh',
+    'csh',
+    'ecmascript',
+    'javascript',
+    'json',
+    'lisp',
+    'rtf',
+    'sh',
+    'tex',
+    'yaml',
+]);
+
+/** @type {(document: Blob) => boolean} */
+export const isPlainText = document => document.type.startsWith('text/') ||
```

Comment:
This is open to iCalendar files (.ics) since they are text/calendar, and Adobe Illustrator files (.aip) since they are text/x-audiosoft-intra. There are probably others.

Reply:
ics files are text, so those don't need to be checked against. I'm not sure about aip files though.. will probably have to check those.